pkg/cloudprovider/cloud.go

```diff
@@ -37,6 +37,8 @@ type Interface interface {
 	Clusters() (Clusters, bool)
 	// Routes returns a routes interface along with whether the interface is supported.
 	Routes() (Routes, bool)
+	// Volumes returns a volumes interface along with whether the interface is supported.
+	Volumes() (Volumes, bool)
```

Comment:
...right up until I add an API in my cluster for provisioning them on the fly.  Having an iSCSI server with an admin API that makes new devices seems totally plausible.

There is not a 1:1 association of cloud-provider to volume plugin and I don't think probing volumes is the right way to figure out which to instantiate